Fix: On Mobile, `X` button on user profile modal does not work

niranjan-uma-shankar opened the pull request:
On mobile viewports, the close (`×`) icon is unresponsive. It also seems redundant, as the modal already provides a "Close" button in the footer. 

This PR fixes the issue by removing the `X` button from the header section of the modal, thereby removing the redundancy. Users can close the modal using the "Close" button in the footer. This fixes one of the issues described in https://github.com/TryGhost/Ghost/issues/24509.

| BEFORE | AFTER 
| ---- | ---
| <video src="https://github.com/user-attachments/assets/61bb40d9-a785-4214-8f7d-67d3ecb97320" /> | <img width="305" height="637" alt="Screenshot 2025-07-27 at 12 18 42 PM" src="https://github.com/user-attachments/assets/ec2c2e59-4001-4d46-8192-9b359c0f8f3b" />

### Testing Instructions

1. Open Ghost Admin in a mobile viewport (DevTools or actual device).
2. Click on "Your profile" ([screenshot](https://cloudup.com/cxziBYfeJGC)).
3. In the profile modal, verify that it does not have the `X` button.

niranjan-uma-shankar committed:
On the user profile modal, replaces the X button with an empty span element.

This fixes an issue with an unresponsive X button on the user profile modal, seen in mobile screens. The X button is replaced with an empty <span> element, since the modal already has a close button in the footer.

niranjan-uma-shankar committed:
Replaces empty <span> element with a fragment

React fragment helps avoid adding an empty <span> element to the DOM.

peterzimon commented on apps/admin-x-settings/src/components/settings/general/UserDetailModal.tsx:
```diff
@@ -417,6 +420,7 @@ const UserDetailModalContent: React.FC<{user: User}> = ({user}) => {
             size={canAccessSettings(currentUser) ? 'md' : 'bleed'}
             stickyFooter={true}
             testId='user-detail-modal'
+            topRightContent={topRightContent}
```
This could be simplified by directly injecting `{<></>}` to `topRightContent.

niranjan-uma-shankar replied:
@peterzimon This update has been made in [813d0d4](https://github.com/TryGhost/Ghost/pull/24510/commits/813d0d45d0ec896a462eedd13d5ff0bc5a788500).

niranjan-uma-shankar committed:
Refactors to inject the fragment directly in the prop

This simplifies the code by injecting the fragment in the prop and removes the const variable declaration.

niranjan-uma-shankar committed:
Adds a helpful code comment

9larsons committed:
Changed user detail modal to use Modal's hideXOnMobile prop

ref https://github.com/TryGhost/Ghost/pull/24510

- the design-system Modal already provides hideXOnMobile for hiding the
  mobile-only close icon (used by stripe-connect-modal and preview-modal),
  so use that instead of overriding topRightContent with an empty fragment
